spec/services/hyrax/collections/nested_collection_query_service_spec.rb

```diff
@@ -214,142 +210,168 @@
         end
 
         describe 'it prevents circular nesting' do
+          before do
+            allow(scope).to receive(:can?).with(:read, coll_c).and_return(true)
+          end
+
           it 'returns an array of collections of the same collection type excluding the given collection' do
-            expect(scope).to receive(:can?).with(:read, coll_c).and_return(true)
-            expect(described_class).to receive(:query_solr).with(collection: coll_c, access: :deposit, scope: scope, limit_to_id: nil, nest_direction: :as_parent).and_call_original
-            expect(subject.map(&:id)).to contain_exactly(coll_a.id, another.id)
+            expect(described_class.available_parent_collections(child: coll_c, scope: scope))
+              .to contain_exactly(have_attributes(id: coll_a.id),
+                                  have_attributes(id: another.id))
           end
         end
       end
     end
   end
 
   describe '.parent_and_child_can_nest?' do
-    let(:parent) { coll_c }
-    let(:child) { another }
-
-    subject { described_class.parent_and_child_can_nest?(parent: parent, child: child, scope: scope) }
-
     before do
       coll_e
       another
       wrong
     end
 
-    describe 'given parent and child are nestable' do
+    describe 'when parent and child are nestable' do
       describe 'and are the same object' do
-        let(:child) { parent }
-
-        it { is_expected.to eq(false) }
+        it 'is false' do
+          expect(described_class.parent_and_child_can_nest?(parent: coll_c, child: coll_c, scope: scope))
```

Comment:
I think this is a typo:
```suggestion
          expect(described_class.parent_and_child_can_nest?(parent: coll_c, child: another, scope: scope))
```